main.nf

```diff
@@ -69,56 +100,84 @@ process generate_folds{
 K_FOLD_CELL_INDEXES
     .flatten()
-    .map{ f -> tuple("${f.simpleName}", f) }
-    .set{FOLDS_BY_K}
-	
+    .map{ f -> tuple("${f}".split("\\.")[1], f) }
+    .set{FOLDS}
+
+// add fold name to data channels
+FOLD_DATA = FOLDS.combine(FETCH_DATA_UNMELT)
+
 // split data into train and test sets based on indices
 process split_train_test{
-	//publishDir "${params.output_dir}/${dataset_id}/split_data", mode: 'copy'
+	publishDir "${params.output_dir}/${dataset_id}/split_data", mode: 'copy'
 	conda "${baseDir}/envs/r-caret.yaml"
 	errorStrategy { task.exitStatus == 130 || task.exitStatus == 137  ? 'retry' : 'finish' }
 	maxRetries 5
 	memory { 16.GB * task.attempt }
 	
 	input:
-	    set val(fold), file(k_fold_cell_indexes) from FOLDS_BY_K 
-	    file(data_dir) from QUERY_10X_DIR 
-	    file(sdrf) from UNMELTED_SDRF
-	    val(dataset_id) from DATASET_ID 
+	tuple val(fold), file(indx), file(data), val(dataset_id), val(barcode_col), val(cell_type_col), val(matrix_type), val(num_clust) from FOLD_DATA
+	
 	output:
-	    set val("${fold}"), file("test.zip"), file("train.zip") into SPLIT_DATA 
+	set val(fold), val(dataset_id),  file("${dataset_id}.test.${matrix_type}.${fold}.zip"), file("${dataset_id}.train.${matrix_type}.${fold}.zip") into SPLIT_DATA
+	set val(fold), val(dataset_id) into FOLD_N_SPLIT_DATA
 	"""
 	split-train-test-data.R\
-		--input-matrix "${data_dir}/matrix.mtx"\
-		--input-barcodes-tsv "${data_dir}/barcodes.tsv"\
-		--input-genes-tsv "${data_dir}/genes.tsv"\
-		--input-cell-indexes ${k_fold_cell_indexes}\
-		--input-processed-sdrf ${sdrf}\
-		--output-test-dir test\
-		--output-train-dir train
-	#zip files by test or train
-	zip -r test.zip test
-	zip -r train.zip train
+		--input-matrix "${data}/10x_data/matrix.mtx"\
+		--input-matrix-type ${matrix_type}\
+		--dataset-id ${dataset_id}\
+		--input-barcodes "${data}/10x_data/barcodes.tsv"\
+		--input-cell-indexes ${indx}\
+		--input-unmelt-sdrf "${data}/unmelted_sdrf.tsv"\
+		--fold-n ${fold}
+	# mv genes and markers files
+	parallel cp -v "${data}/10x_data/genes.tsv" ::: ${dataset_id}.test.${matrix_type}.${fold}/10x_data ${dataset_id}.train.${matrix_type}.${fold}/10x_data
+	parallel cp -v "${data}/marker_genes_${num_clust}.tsv" ::: ${dataset_id}.test.${matrix_type}.${fold}/ ${dataset_id}.train.${matrix_type}.${fold}
+	# zip files by test or train
+	zip -r ${dataset_id}.test.${matrix_type}.${fold}.zip ${dataset_id}.test.${matrix_type}.${fold} 
+	zip -r ${dataset_id}.train.${matrix_type}.${fold}.zip ${dataset_id}.train.${matrix_type}.${fold} 
 	"""
 }
 
-// Run cell-types-eval-control-workflow which runs the predictor tools
-process run_cell_types_eval {
+N_UNIQUE_MATRICES = N_UNIQUE_MATRIX_TYPES.count()
+// group data by fold (and dataset_id) and flatten tuple  
+GROUPED_DATA = SPLIT_DATA.groupTuple(by:[0, 1]).flatten().buffer(size: 4, skip:2)
+
+// move same fold data into single dir
+process group_fold_data {
+	input:
+	file('*') from GROUPED_DATA
```

Comment:
You don't actually need to use `*` as a file name, as there's no literal matching. You could just use `file`, for that matter.